File: Container With Most Water - Leetcode 11/Container With Most Water - Leetcode 11.java
public class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int l = 0;
        int r = n - 1;
        int maxArea = 0;

        while (l < r) {
            int w = r - l;
            int h = Math.min(height[l], height[r]);
            int a = w * h;
            maxArea = Math.max(maxArea, a);

            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }

        return maxArea;
    }
}

File: Container With Most Water - Leetcode 11/Container With Most Water - Leetcode 11.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let l = 0;
    let r = height.length - 1;
    let maxArea = 0;

    while (l < r) {
        const w = r - l;
        const h = Math.min(height[l], height[r]);
        const a = w * h;
        maxArea = Math.max(maxArea, a);

        if (height[l] < height[r]) {
            l++;
        } else {
            r--;
        }
    }

    return maxArea;
};
